### DISCUSSION TITLE
And another one getting started!

### DISCUSSION THREAD

**Discussion by unhuzpt** [Short-bus/pilomar#72] [General]
Matt, a BIG thanks from me! :)

Amazing project, instructions and support.
So i am half way building it and loving each step of the way.
Printed all in PETG-CF and each piece is amazing!

Pretty sure in some days my questions and help requests will flood in :)

Cheers!

**Comment by unhuzpt** [comment]
![current](https://github.com/Short-bus/pilomar/assets/100961314/462d5ffb-b6b7-4734-907f-904a2ea96ef2)

Honestly i have a few doubts about the seating of the round parts in the 3 bearings, looking at your video and the feel i get seems there is too much slack on the round part between the bearings, is like it moves about 1 or 2mm around. Is this expected or the bearings should make this part snug fit between them?

**Reply by Short-bus** [reply]
Do you mean the 'R' Altitude Hub that sits on the three bearings in the camera tower?
It's unusual for someone to find them slack - usually they require a bit of sanding to move smoothly.
It may not be a big problem though. When you add the altitude motor and gear you can increase the pressure on the altitude hub. That will probably reduce the slack. 
Try that first, if the slack remains after that then we may need to get more creative! Hopefully it works for you.
It might be worth measuring the outside diameter of the bearings too. Just in case there is some unexpected variation there too.
I would expect them to be 19mm outer diameter.
Good luck!

**Reply by unhuzpt** [reply]
Thank you for your reply!
In fact the bearings outer diameter is 20mm - what is even more surprising and the diameter of the piece that they slide in (R.13.3) is 61mm.
I did not have to sand the surface where the bearings are to slide in, but in your video you mention the R.13.3 piece should slide snug on the bearings and i have that slack. 
I ordered the wrong collars so now i am kind of stuck waiting for them to arrive - i used 6mm shaft and the collars even if 6mm do not slide in so i ordered 6,1mm inner diameter ones.

**Reply by Short-bus** [reply]
61mm sounds good, the CAD diagram shows the same. Hopefully fitting the drive mechanism will fix it anyway.
![image](https://github.com/Short-bus/pilomar/assets/84185921/d0780496-3f18-48b4-b553-ed3ca389d64e)

**Reply by unhuzpt** [reply]
Thank you for your reply Matt!
I will see how it comes, i have never done this kind of project so yeah... i am more or less comfortable with electronics and the software part, but my construction skills are really bad, Have a great day!

**Comment by unhuzpt** [comment]
And progress is ongoing :)
![board_01](https://github.com/Short-bus/pilomar/assets/100961314/75c8497e-b58c-4e79-b86a-e0d7f5067d7f)

**Reply by unhuzpt** [reply]
Hello Matt

Got back to the issue on the altitudehub i had with too much slack between the 3 bearings and after some calculations ended up scaling the part to 101,50% so now i have it snug and spinning like a dream. :)
Also decided to put some lithium grease on the path the bearings travel so that is kept lubed and smooth.

Cheers
Fred

**Reply by Short-bus** [reply]
> Got back to the issue on the altitudehub i had with too much slack between the 3 bearings and after some calculations ended up scaling the part to 101,50% so now i have it snug and spinning like a dream. :)

Brilliant idea Fred, well done. 

I've got a wider tower design in the works this year. Based on your feedback I've changed the top bearing bolt hole into a slot, so in future the fit can be adjusted.

Matt

**Reply by unhuzpt** [reply]
Hi there Matt

Great to hear this!
I have seen you seek to use bigger lens and are working on a new iteration of the project.
I hope i can make mine work before this happens :) 
Basic electronics understanding and a dream of watching the skies but never had possibility in the past i guess will lead me to the expected goal. 
Has been amazing as i have my 9 year old son engaged on this also, my aim is to open his appetite and mind to tall involving this project.

At this rate i should have my telescope in test in 2 weeks.
Waiting for the last bits (in fact just the power connectors to the pcb and then hope i do not mess the electrics :)

PS-i am printing in a bambu labs p1s and is pretty much dialed in and parts accuracy are quite good. Strange more people did not run into this issue.

**Reply by unhuzpt** [reply]
Cheers Matt

Does this pass your visual inspection?
![IMG-20240525-WA0002](https://github.com/Short-bus/pilomar/assets/100961314/e660e611-712d-46d8-b356-f66dc35f34c8)
![IMG-20240525-WA0004](https://github.com/Short-bus/pilomar/assets/100961314/2636bd83-d1d5-497e-bc73-87001b1783c5)
![IMG-20240525-WA0000](https://github.com/Short-bus/pilomar/assets/100961314/806fcc34-4a18-4aa6-82a9-61a3f86f4e18)

**Comment by unhuzpt** [comment]
Now comes the issues for me :)

- best solution to power both motors and pi
  can i just use the pi power supply in an extension cord and another power supply i have (used for led strip) connected on the same extension   cord?

- how to verify and set the motor drivers voltage? Is that needed? If so what is it?

**Reply by Short-bus** [reply]
Hi,

1) Power supplies: Yes I think you'll be fine, I use a single extension cable with 2 sockets. I put the RPi power supply in one, and the motor power supply in the other. I've not had a problem with them. I use an outdoor 'rainproof' extension cover just to protect things from humidity or surprise rain!
2) Motor voltage: As I understand it, stepper motors are more concerned with the current being right. If you've got a 12V motor and a 12V power supply then focus on the current settings. To set the current you use the little potentiometer on the DRV8825 chips. There are instructions online which walk you though the process, if you've a voltmeter and a small screwdriver you can do it quite easily. I set the current limit to about 0.5A to start with. If the motors are working well, leave it at that. You can increase the current later on if you see the benefit.
Remember that each motor has 2 coils. The 0.5A is 'per coil'... so the power supply has to deliver 0.5A x 2 coils x 2 motors = 2A. If you increase the current to the full 2A then you need 2A x 2 coils x 2 motors = 8A. I use a 10A power supply to make sure I am always within limits.
The Pololu website has information about setting current limits on their DRV8825 product page.
https://www.pololu.com/product/2133
And it links to a detailed video showing the process too.

The build is looking really nice, well done. The print quality is really impressive.
Matt

**Reply by unhuzpt** [reply]
Hello Matt

Thank you for your kind reply!
I have gotten a 10A power supply also to have things on the safe side.
At the moment i am "blocked" because after putting all in place i noticed i need a L shaped usbc male to female adapter otherwise it does not fit due to the control board too close :)
So while i wait for it to arrive i am going to triple check everything, proper cable routing and get ready to fire it up for the first time.

Thank you for the hints and link on the stepper motor setting!
Keep you posted.

Have a fantastic week!

**Reply by Short-bus** [reply]
I noticed a similar problem fitting the HDMI cable back onto the RPi. I added some spacers behind the RPi mounting board to move it 10mm further from the tower. That gave more room below for cables to be fitted. You could use longer bolts and some extra nuts to position the board where you need. I stacked a couple of spare 'DA' nut holders on each bolt.

![image](https://github.com/Short-bus/pilomar/assets/84185921/5234bd6e-5f42-4d4a-bd14-d6a1bb56c8b7)

![image](https://github.com/Short-bus/pilomar/assets/84185921/3243837a-8068-4ece-84e2-12b5e53d400e)

**Reply by unhuzpt** [reply]
Hi Matt

Thanks for this great idea. I may end up doing it also.
But for now i will try to keep as is because i only secured the 2 lower bolts on the pi support part and used the motor control board to have it seated on to. May end up going your route :)

By the way, when you want to take it away from home, what kind of power source you use?
Like a car battery and a dc converter for the pi?

**Reply by Short-bus** [reply]
### Battery power

I haven't run mine off a battery yet, though I think some other people have. I guess a car battery should work with a DC convert for the RPi. You need just under 2A for a fully loaded RPi4B I think, it can run quite hard at times and needs to provide a little bit of extra current for the microcontroller. Then you need at least 2A for the motors. So any battery which can provide about 4A+ reliably for a few hours is a good start. 
You should also consider how to control the system, you'll need a wifi-hotspot of some form (phone?) to access the command line. Or you need to connect a keyboard and monitor to the RPi.
I should try this too and see if there are other improvements to make to support 'wild' operation. 

(A web UI provided directly over wifi would be ideal)

**Comment by unhuzpt** [comment]
Ok....
So i went and powered on the circuit....

Let's start by the positive things:
- pi boots and software is currently running the catalog optimization.
- nothing exploded, no fire, no strange smell
- i think i created a extra terrestrial antenna!

And the not so positive things:
- without pi connected to controller board and applying the 12v dc and motors connected i had 0 volts on the stepper drivers (measuring between ground of stepper and pot to tune the voltage
- i thought i had to have the tiny doing something for the motor to do something.....
- next powered the pi also and my motors became a beacon for extraterrestrial signals! :)
  they sound like they are picking static and sometimes beeps.... and on the stepper drivers i read 1,6v and no change on voltage if i rotate    their little pot.....

So i imagine they using a lot more power than expected and i have some problem somewhere....
I don't see voltage in any of the 2 resistors. The tiny flashes blue/purple i think form time to time....
No idea

Help! :)

**Reply by Short-bus** [reply]
If there's no power to the RPi (& Tiny) the DRV8825s probably don't get enabled, so nothing will happen on the motors.
When you power up the RPi I think the DRV8825s wake up, then the motors get energised.
They should be quite quiet if they are not moving - just a small hum.
If they are very noisy, check that the 4 wires are connected correctly to the PCB perhaps? A+/A- B+/B-
0V on the potentiometer certainly sounds wrong.

But it is best to check that the Tiny is working OK first.

I suggest disconnecting the 12V power to the motors and verify the Tiny is happy before trying to move the motors.

There should be about 3.3V between the LEFT of resistor R1 and GND if the RPi has turned the Tiny on.
There should be about 2.5V between the Tiny's 5V pin and GND when it's on too.
![image](https://github.com/Short-bus/pilomar/assets/84185921/2db9fccc-4c75-4933-aabd-78cd0390bbf2)

The 'blue/purple' flash sounds strange. After the Tiny has booted I would expect to see a short blue flash every few seconds, but no other colors. Purple suggests something else is happening.

On the _microcontroller tools_ menu you can select _Monitor Communication_. 

Check that there are messages to/from the microcontroller appearing in that display.

You should see something like this...

```
RPi received: motor status 20240527212253 azimuth n 20240527212253 0 192000 180.0 y n 0.1 0 tmr
RPi received: motor status 20240527212253 altitude n 20240527212253 0 0 0.0 y n 0.1 0 tmr
RPi received: session status 20240527212259 y n y 927 0 tmr 0
RPi received: comms status 20240527212259 0 973 22168 0 17308 0 1 tmr
RPi received: motor status 20240527212303 azimuth n 20240527212303 0 192000 180.0 y n 0.1 0 tmr
RPi received: motor status 20240527212303 altitude n 20240527212303 0 0 0.0 y n 0.1 0 tmr
RPi queueing (Q# 0): # heartbeat [95]
RPi received: motor status 20240527212313 azimuth n 20240527212313 0 192000 180.0 y n 0.1 0 tmr
RPi received: motor status 20240527212313 altitude n 20240527212313 0 0 0.0 y n 0.1 0 tmr
```

If there is no communication at all the RPi will eventually print an error message and try to restart the Tiny. 

If that's OK, then try starting an observation. Keep the 12V disconnected, the software will still work even with no power to the motors.

If the system is happy it will start issuing messages as it captures images. If there is anything wrong with communication it will complain that the motors are not configured.

When you eventually shut down the software give 'Y' to the Home Motors question. It will reset everything ready for when you add 12V power later.

If all that works OK then we know that RPi, comms and Tiny are happy. 
Then you can investigate the motor power with more confidence.

Which means working out why there's 0V on the potentiometer first. For safety disconnect the 4 wires for each motor in case it's delivering crazy current to them. Then reconnect the 12V supply to the motorcontrol board and try again to get a voltage on the potentiometer with the pilomar program sitting at the main menu.

You should have 12V between VMOT and GND on each of the DRV8825s.
![image](https://github.com/Short-bus/pilomar/assets/84185921/2a920558-2d7a-4ffa-a5e7-84d4ea3b5477)

Also measure the voltage across A+/A- and B+/B- terminals where the 4 motor wires would connect.
![image](https://github.com/Short-bus/pilomar/assets/84185921/35a98b2e-24e5-4616-b4ea-b0b711082260)

Run those tests and we'll see what we have uncovered.

_BTW: I was checking your build images while thinking about this and noticed that the camera sensor may be mounted upside down. The tripod mount hole should face DOWN not UP._

**Reply by unhuzpt** [reply]
Thank you so much for you detailed response! If i knew you would reply so fast i would have done all this yesterday,
I am not going to stat troubleshooting as per your instructions and let's see.
For reference here is a pic of the motor wiring, i followed the diagram in the instructables pdf assuming we using the very same hardware.
The software seems to be working fine.
![motor_wiring](https://github.com/Short-bus/pilomar/assets/100961314/3e825bee-f931-4e9a-b78f-f246debcc6aa)

Thank you for your time and help.

**Reply by unhuzpt** [reply]
Hello Matt!

So far so good all seems to work as expected. Now going to the motors part :)

![Screenshot 2024-05-28 063345](https://github.com/Short-bus/pilomar/assets/100961314/6a7ebe0c-5c92-4117-8ca3-14b23c8d4400)


![Screenshot 2024-05-28 064934](https://github.com/Short-bus/pilomar/assets/100961314/3df726d0-feb5-4dd3-9c95-e296a2deeb5f)

**Reply by unhuzpt** [reply]
Ok....
So i get 1.6v on the stepper motor board and seems my problem is ....
I just can't make the pot to turn..... none of my keys seem to make the god dam thing to move.

**Reply by unhuzpt** [reply]
Ok!
Managed to set 0.5v on the steppers and now my setup does not seem a alien phone boot anymore!
Have 12v in each coil.

But now i realized we need to install the camera and i am a bit lost here...

![image](https://github.com/Short-bus/pilomar/assets/100961314/6db6990e-2159-4414-9906-5a9ef03c13db)

How you managed to install in buster?

**Reply by Short-bus** [reply]
Great that you've got the DRV8825s setup. Well done. Those little pots are fiddly, I have a favourite screwdriver just for that job!
The screenshot for the observation test looks correct too. It seems happy.

If you've installed BUSTER then it comes with the raspistill camera libraries already. The software should recognise that and work with it. You don't have to install picamera2

If you've installed BOOKWORM then it comes with libcamera libraries - which includes picamera2 automatically.
So hopefully you don't have to install anything extra for the cameras

Things to consider:
- When you install the operating system always install the FULL image, including the 'desktop'. It provides useful services that make things easier.
- You can add libcamera to Buster, there are instructions online, but I've not tried it. It is easier to install a Bookworm O/S instead.
- If you build with Bookworm, it's the 64bit O/S and is faster and more complete than the old Buster 32bit one. The telescope performs more smoothly with Bookworm. (The next version of the software I'm working on has more features with Bookworm too.)
- If you install Bookworm you should use the /scripts/buildpilomar_bookworm64 script. It's simpler and faster than the original Buster one too!

**Reply by unhuzpt** [reply]
Thank you so much for all this info and support.
Yeah things seem to work as expected.... well sort of but is a problem between chair and keyboard.
I have to revisit all the positioning of the gears and probably also the current on the motors as the azimuth turns but makes a lot of noise and i see the motor mount wiggle - maybe to tight to the worm gear - and the altitude does not even turn, in spite of the motor best efforts :)

And yeah, i go to re-build with bookworm as indeed i installed buster.

Getting very exited with this!

**Reply by Short-bus** [reply]
Great progress, well done.
Hopefully there's plenty of adjustment for the gears you should be able to get it running smoothly.

By default the motors run in 'full steps' which are big and noisy. When you are happy that the gears are turning smoothly you can change the parameters to use 'microstepping' which will make the movement smoother and quieter.

**Reply by unhuzpt** [reply]
Ahhh! :)

There was one point you mentioned and I keep forgetting to reply and address:
The position of the camera, I turned it around as i could not find a way to have in up position the thingies to rotate  (now you see i know a lot about camera lenses....). 
You mention in instructions to put in infinite focus.... that is something i still need to learn about what it really means but yeah.
Don' t want to bother you with too much questions, at least I hoped for  :)

**Comment by Short-bus** [comment]
### Focussing on infinity
The little screwdriver that comes with the Hi Quality sensor is really useful for reaching into the space and adjusting the back focus ring.
With practice I found a couple of things that make focusing easier.

- You can remove the camera from the telescope and mount it on a regular tripod. That makes it very easy to access and adjust focus.
- Set the LENS to infinity. Put some tape on the focus ring. You don't want it to move again. 
- Then release the 'back focus ring' lock screw on the sensor using the little screwdriver. This has a fine screw thread and lets you make very fine adjustments.
- Take photographs of any bright star and keep adjusting the back focus ring until it is as sharp as you can get.
- Retighten the back focus ring lock screw with the little screwdriver.
- It's a bit of an art and takes patience.

![image](https://github.com/Short-bus/pilomar/assets/84185921/22e7a486-8f6f-4bae-9cb7-d99e6ca3734f)

I sometimes have to adjust the focus again after a few months, but it should stay sharp for a long time. 

The next version of the camera tower will try to make this easier to do, I might mount the camera(s) in portrait orientation rather than landscape so that the adjustment screw is VERY easy to access.

**Reply by unhuzpt** [reply]
> ### Focussing on infinity
> 
> The little screwdriver that comes with the Hi Quality sensor is really useful for reaching into the space and adjusting the back focus ring. With practice I found a couple of things that make focusing easier.
> 
>     * You can remove the camera from the telescope and mount it on a regular tripod. That makes it very easy to access and adjust focus.
> 
>     * Set the LENS to infinity. Put some tape on the focus ring. You don't want it to move again.
> 
>     * Then release the 'back focus ring' lock screw on the sensor using the little screwdriver. This has a fine screw thread and lets you make very fine adjustments.
> 
>     * Take photographs of any bright star and keep adjusting the back focus ring until it is as sharp as you can get.
> 
>     * Retighten the back focus ring lock screw with the little screwdriver.
> 
>     * It's a bit of an art and takes patience.
> 
> 
> ![image](https://private-user-images.githubusercontent.com/84185921/334355823-22e7a486-8f6f-4bae-9cb7-d99e6ca3734f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4OTIzNjQsIm5iZiI6MTcxNjg5MjA2NCwicGF0aCI6Ii84NDE4NTkyMS8zMzQzNTU4MjMtMjJlN2E0ODYtOGY2Zi00YmFlLTljYjctZDk5ZTZjYTM3MzRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI4VDEwMjc0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMWI5MWQ1Y2U4YTU5YzdkMzM5NTJhYTNmYjk1MWU3YjI5OWRmZGRmOTZmZTA1N2NjMGM1YTQwZDI0YjIyMWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pze9qJGHETQQBLJhqrD7bPiFre7oA_Tu7in96RtZ4TE)
> 
> I sometimes have to adjust the focus again after a few months, but it should stay sharp for a long time.
> 
> The next version of the camera tower will try to make this easier to do, I might mount the camera(s) in portrait orientation rather than landscape so that the adjustment screw is VERY easy to access.

You are the BEST! 
Thanks a LOT!

**Reply by unhuzpt** [reply]
Is this something you already know about?
I am running in bookworm with full install :)

`Camera enabled
2024-05-28 10:25:28 No recovery log file found for azimuth motor. Assuming  180.000Â°
2024-05-28 10:25:28 No recovery log file found for altitude motor. Assuming  000.000Â°
2024-05-28 10:25:28 Home location Latitude 49.781880 N, Longitude 6.225176 E
2024-05-28 10:25:28 Hipparcos data cache exists, using that.
[#################################] 100% de421.bsp
2024-05-28 10:25:35 Loading Messier catalog from /home/unhuzpt/pilomar/data/messierobjects.json ...
[#################################] 100% constellationship.fab
2024-05-28 10:25:36 Loading constellation patterns...
2024-05-28 10:25:36 Loading New General Catalog (NGC) entries from /home/unhuzpt/pilomar/data/ngc.json ...
2024-05-28 10:25:36 No NGC data cache, creating one now.
Exception in thread Thread-1 (StartMctlComms):
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/home/unhuzpt/pilomar/src/pilomar.py", line 3472, in StartMctlComms
    Mctl.CommsLoop(UartControlQueue)
  File "/home/unhuzpt/pilomar/src/pilomar.py", line 3404, in CommsLoop
    self.ReadPoll() # Read anything waiting in the input buffer.
    ^^^^^^^^^^^^^^^
  File "/home/unhuzpt/pilomar/src/pilomar.py", line 3226, in ReadPoll
    while self.uart.in_waiting: # Something in the read queue.
          ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/serial/serialposix.py", line 549, in in_waiting
    s = fcntl.ioctl(self.fd, TIOCINQ, TIOCM_zero_str)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 5] Input/output error
2024-05-28 10:26:12 Loading meteor shower list from /home/unhuzpt/pilomar/data/meteors.json ...
2024-05-28 10:26:12 Loading comet list from https://www.minorplanetcenter.net/iau/MPCORB/CometEls.txt ...
[#################################] 100% CometEls.txt
2024-05-28 10:26:14 Loading CelesTrak station data from https://celestrak.org/NORAD/elements/gp.php?GROUP=stations&FORMAT=tle
You must select a target to start the program.

 Select target 
 1 Resume last observation      2 Repeat earlier observations 
 3 Solar system object          4 Hipparcos star catalog      
 5 Messier catalog              6 New General Catalog (NGC)   
 7 Comet                        8 Meteor shower               
 9 Space stations/satellites   10 RA-DEC co-ordinates         
11 Fixed ALT-AZ point          
 x Exit                         ? Refresh                    
Menu option : 


`

**Reply by Short-bus** [reply]
That's a new one.

Does it occur again if you restart the program?

I wonder if it's similar to this issue...
https://github.com/Short-bus/pilomar/issues/39 **After building the Hipparcos star catalog, pilomar terminates at target prompt**

**Reply by unhuzpt** [reply]
Hi there Matt!

So......
Yeah, the "issue" was that your script assumes the user is 'pi' and that was not my case :)
So tired lately that i forgot this when i build the 1st time and noticed it....
Now it installed all fine and is working as expected :)

Allow me to suggest maybe to adapt the script to something like:

#Get current user
username=$(whoami)

#use current user
export pilomarhome="/home/${username}/pilomar"

Or maybe even ask if one wants to use the current user or another?

**Reply by Short-bus** [reply]
Doh! It should work with different users, so there's a bug somewhere.
It looks at the runtime location and should establish the file structure based upon that.
I'll need to check out what's happening there.
The trace initially looked like it was a UART error.

Do you have the log file still from that case? /home/..../log/pilomar_2024052810_????_.log

I've raised it as issue #74
